earthly/mithril_snapshot/Earthfile

```diff
@@ -1,19 +1,86 @@
-VERSION 0.7
+VERSION 0.8
 
 # cspell: words preprod
 
-package-preprod-snapshot:
-    FROM alpine:3.19
+base-snapshot-builder:
+    # Note: an official docker image is here, but it could not reliably pull from Earthly.
+    # It gives permission denied errors randomly.
+    # Image comes from: https://github.com/input-output-hk/mithril/pkgs/container/mithril-client
+    # FROM ghcr.io/input-output-hk/mithril-client:latest
+
+    # Build our own simple mithril client image
+    FROM debian:stable-20240408-slim
+
+    RUN apt-get update && apt-get -y install wget jq zstd
+
+    RUN wget -O mithril-cli.deb \
+            https://github.com/input-output-hk/mithril/releases/download/2412.0/mithril-client-cli_0.7.6+2e462c0-1_amd64.deb
+    RUN dpkg -i mithril-cli.deb
+
+DL_SNAPSHOT:
+    FUNCTION
+    ARG network
+    ARG mithril_network
+    ARG snapshot_digest=latest
+    ARG validate=false
+
+    LET NETWORK=$network
+    LET AGGREGATOR_ENDPOINT=https://aggregator.${mithril_network}.api.mithril.network/aggregator
+    LET GENESIS_VERIFICATION_KEY=$(wget -q -O - https://raw.githubusercontent.com/input-output-hk/mithril/main/mithril-infra/configuration/${mithril_network}/genesis.vkey)
+    LET SNAPSHOT_DIGEST=$snapshot_digest
 
     WORKDIR /snapshot
 
-    RUN wget -O \
-        preprod.tar.zst \
-        https://storage.googleapis.com/cdn.aggregator.release-preprod.api.mithril.network/preprod-e135-i2625.80b0c02cd0c86c47a79a9b6d0d9b0e6936d5f621e8094ed53405fef3ee103a3b.tar.zst
+    IF $validate
+        # Validation is slower, but the file will be guaranteed to be valid.
+        # This automatically handles unarchiving the snapshot
+        RUN mithril-client -vvv cardano-db download --json $snapshot_digest
+    ELSE
+        # This is faster, but it could be a faked snapshot file.
+        LET DLFILE=$(mithril-client cardano-db snapshot show --json $snapshot_digest | jq '.locations[0]' | tr -d '\"')
+        RUN mkdir -p ./db && \
+            wget -O - $DLFILE | zstdcat | tar -xf - -C ./db
+    END
+
+mainnet-snapshot:
```

Comment:
WDYT to make all these targets which are produce artefacts as `package-*` ?
So it cat-ci we can see that they are produced correctly, otherwise ci for this repo will no trigger these targets

Reply:
Because i don't want mainnet to run in CI at all.
We can add a separate package target, which I already have to do it for preprod which we will use in CI.
The other targets are a developer convenience only at this point.

Reply:
@saibatizoku has a mithril tester which runs on preprod snapshot now, but it finishes in < 30 seconds when running in earthly.
I want to test against manner snapshot to see what happens over time, and that why I made this.